src/js/store/scratchOrgs/reducer.ts

```diff
@@ -0,0 +1,40 @@
+import { ScratchOrgsAction } from './actions';
+
+export type ScratchOrg = {
+  canceled_at: null;
+  config: null;
+  created_at: string;
+  edited_at: string;
+  email: string;
+  enqueued_at: string;
+  id: string;
+  job_id: string;
+  plan: string;
+  status: string;
+};
+export type ScratchOrgState = {
+  [key: string]: ScratchOrg;
+};
+
+const reducer = (
+  scratchOrgs: ScratchOrgState = {},
+  action: ScratchOrgsAction,
+): ScratchOrgState => {
+  switch (action.type) {
+    case 'SCRATCH_ORG_SPINNING':
+    case 'SCRATCH_ORG_CREATED': {
+      const org = action.payload;
+      const existingorg = org[org.plan];
```

Comment:
@jgerigmeyer I don't understand the linting error here, but leads me to think there is either something wrong with how scratchOrg types are defined or there needs to be more annotations for the type expected here :shrug: